opm/autodiff/BlackoilModelBase_impl.hpp

```diff
@@ -1476,8 +1608,33 @@ namespace detail {
             std::copy(&rv[0], &rv[0] + nc, reservoir_state.rv().begin());
         }
 
+
+        updateWellState(dwells,well_state);
+
+        // Update phase conditions used for property calculations.
+        updatePhaseCondFromPrimalVariable();
+    }
+
+    template <class Grid, class Implementation>
+    void BlackoilModelBase<Grid, Implementation>::updateWellState(const V& dx,
```

Comment:
The `dx` argument should be renamed `dwells` to avoid confusion (since it is not the same as the dx passed to updateState()).